project/src/components/room/room.jsx

```diff
@@ -1,14 +1,28 @@
 import React from 'react';
+import ReviewsForm from '../reviews-form/reviews-form';
+import PropTypes from 'prop-types';
+import { useParams } from 'react-router-dom';
+import { getStarRaiting } from '../../common';
 
-function Room() {
+function Room(props) {
+  const { offers } = props;
+
+  const params = useParams();
+  const id = params.id;
+
+  const offer = offers.find((element) => String(element.id) === id);
```

Comment:
```suggestion
  const offer = offers.find((element) => element.id === Number(id));
```
Это было бы тоже классно вынести в функцию, и сделать компонент чище